File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        // Max of min of pairs
        // difference of pairs * width should be maximum
        // 2 For loops - O(n**2)
        // Which Pointer to move
        int lo= 0;
        int hi = height.length -1;
        int maxi = 0;
        while(lo<hi){
            int width = hi-lo;
            int hei = Math.min(height[lo], height[hi]);
            maxi = Math.max(maxi, hei*width);
            if(height[lo]<= height[hi]){
                lo++;
            }
            else hi--;
        }
        return maxi;
        
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        lo =0
        hi = len(height)-1
        ma = 0
        wi = 0
        while lo<hi:
            wi = hi-lo
            if height[lo]<=height[hi]:
                ma = max(ma, wi*height[lo])
                lo+=1
            else:
                ma = max(ma, wi*height[hi])
                hi-=1
        return ma
            
        